base/common/python/pki/nssdb.py

```diff
@@ -1145,6 +1145,62 @@ def show_certs(self):
         logger.debug('Command: %s', ' '.join(map(str, cmd)))
         subprocess.check_call(cmd)
 
+    def get_trust(self, nickname, token=None):
+        """
+        Get trust of certificate from NSSDB
+
+        :param nickname: Nickname of the cert
+        :type nickname: str
+        :param token: Token where cert resides
+        :type token: str
+        :return: Trust flag
+        :rtype: str
+        """
+        tmpdir = tempfile.mkdtemp()
+        try:
+            token = self.get_effective_token(token)
+            password_file = self.get_password_file(tmpdir, token)
+            cmd = [
+                'certutil',
+                '-L',
+                '-d', self.directory
+            ]
+            fullname = nickname
+
+            if token:
+                cmd.extend(['-h', token])
+                fullname = token + ':' + fullname
+
+            if password_file:
+                cmd.extend(['-f', password_file])
+
+            logger.debug('Command: %s', ' '.join(map(str, cmd)))
+
+            p = subprocess.Popen(cmd,
+                                 stdout=subprocess.PIPE,
+                                 stderr=subprocess.PIPE)
+
+            output, error = p.communicate()
+
+            if error:
+                # certutil returned an error
+                # raise exception unless its not cert not found
+                if error.startswith(b'certutil: Could not find cert: '):
+                    return None
+
+                raise Exception('Could not find certificate: %s: %s' % (fullname, error.strip()))
+
+            if p.returncode != 0:
+                logger.warning('certutil returned non-zero exit code (bug #1539996)')
+
+            re_compile = re.compile(r'^' + fullname + r'(.*$)', re.MULTILINE)
```

Comment:
I'd suggest to use a proper separator (i.e. whitespaces) between the matching pattern (i.e. fullname) and the text we want to grab to avoid ambiguity, probably something like this:
```
re_compile = re.compile(r'^' + fullname + r'\s+(\S+)$')
```
I think the `$` should not be included in the parentheses, and we should process one line at a time.

Reply:
your regex works. But, as per official doc [1] I need to use `re.MULTILINE` to treat every new line as a separate string.

[Here](https://www.debuggex.com/r/Sgjw8YhnAVmRIWyA) is a quick example i created. Try to remove the `Flags: m` and you should see that there is no match produced.

[1] https://docs.python.org/3.8/library/re.html#re.MULTILINE

Reply:
Ah yes, you're processing the whole output at once instead of line-by-line.